File: 백준/Silver/15828. Router/Router.py
import sys
from collections import deque
input = sys.stdin.readline

n = int(input())
router = deque()

while True:
    info = int(input()) #발생한 시간순서대로 -> 그렇다면 먼저 들어온 친구들은 먼저 나가야함..
    if info == -1:
        break
    if info>0:
        if len(router)<n:
            router.append(info)
    elif info == 0:
        if router:
            router.popleft() #가장 먼저 들어온 패킷 제거

if router:
    print(*router)
else:
    print("empty")

File: 백준/Silver/15828. Router/Router.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine());

        Queue<Integer> router = new LinkedList<>();

        while(true){
            int info = Integer.parseInt(br.readLine());
            if(info ==-1){
                break;
            }
            if(info > 0){
                if(router.size()<n){
                    router.offer(info);
                }
            }else if(info == 0){
                if(!router.isEmpty()){
                    router.poll();
                }
            }
        }
        if(router.isEmpty()){
            System.out.println("empty");
        }else{
            StringBuilder sb = new StringBuilder();
            for(int packet:router){
                sb.append(packet).append(" ");
            }System.out.println(sb.toString().trim());
        }

    }
}
